app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -374,6 +376,23 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
         }
     }
 
+    override fun onResume() {
+        super.onResume()
+
+        lifecycleScope.launch(Dispatchers.Cached) {
+            // We are delaying the read of the clipboard a bit, because ClipboardManager will return
+            // null if the does app not have input focus and that seems to be happening here on
+            // non-first resumes.
+            @Suppress("MagicNumber")
+            delay(500)
```

Comment:
This delay is a bit ugly. But on resume after the app was already running I got `null` from `ClipboardManager` (except for the very first resume when the screen is displayed for the first time). Visually the delay is no problem, but I can't be fully sure it is good enough for all devices.

Reply:
@pocmo OK I was able to reproduce the intermittent `null` from `ClipboardManager` but it seems to also be fixable by doing:

```kotlin
view?.post {
  lifecycleScope.launch(Dispatchers.Cached) {
    store.dispatch(
      SearchFragmentAction.UpdateClipboardUrl(
        requireContext().components.clipboardHandler.url
      )
    )
  }
}
```

Wdyt?

Reply:
This will queue the code to run shortly after the resume. Hm, if that works consistently then we may be able to launch the coroutine on the main thread and switch the context. I'll try that too.

Reply:
That didn't work reliable. But the `post` does. I'll update the PR and then land it.